libnopegl/src/backends/gl/pipeline_gl.c

```diff
@@ -471,8 +427,11 @@ static int pipeline_graphics_init(struct pipeline *s)
     if (gl->features & NGLI_FEATURE_GL_VERTEX_ARRAY_OBJECT) {
         ngli_glGenVertexArrays(gl, 1, &s_priv->vao_id);
         ngli_glBindVertexArray(gl, s_priv->vao_id);
-        init_vertex_attribs(s, gl);
+    } else if (gl->features & NGLI_FEATURE_GL_OES_VERTEX_ARRAY_OBJECT) {
+        ngli_glGenVertexArraysOES(gl, 1, &s_priv->vao_id);
+        ngli_glBindVertexArrayOES(gl, s_priv->vao_id);
     }
```

Comment:
can you add the else assert in this `pipeline_graphics_init` function (and maybe in `bind_vertex_attribs`), like you did in the `ngli_pipeline_gl_update_attribute` function?

Reply:
Done in both commits.